unity-client/Assets/Scripts/MainScripts/DCL/Controllers/Settings/SettingsControllers/AutoQualitySettingsComponent.cs

```diff
@@ -17,24 +17,44 @@ public class AutoQualitySettingsController : MonoBehaviour
     internal QualitySettingsData qualitySettings => Settings.i.autoqualitySettings;
 
     internal int currentQualityIndex;
-    internal IAutoQualitySettingsEvaluator evaluator = new AutoQualitySettingsEvaluator();
     private Coroutine settingsCheckerCoroutine;
+    internal IAutoQualityController controller;
+    internal bool fpsCapped;
 
-    void Awake()
+    void Start()
     {
         if (autoQualityEnabled == null || qualitySettings == null || qualitySettings.Length == 0)
             return;
 
         currentQualityIndex = PlayerPrefs.GetInt(LAST_QUALITY_INDEX,(qualitySettings.Length - 1) / 2);
+
+        controller = new AutoQualityUncappedFPSController(currentQualityIndex, qualitySettings);
+
+        Settings.i.OnQualitySettingsChanged += OnQualitySettingsChanged;
         autoQualityEnabled.OnChange += SetAutoSettings;
+
+        fpsCapped = !Settings.i.currentQualitySettings.fpsCap;
+        OnQualitySettingsChanged(Settings.i.qualitySettings);
         SetAutoSettings(autoQualityEnabled.Get(), !autoQualityEnabled.Get());
     }
 
     private void OnDestroy()
     {
-        if (autoQualityEnabled == null)
+        if(autoQualityEnabled != null)
+            autoQualityEnabled.OnChange -= SetAutoSettings;
+        Settings.i.OnQualitySettingsChanged -= OnQualitySettingsChanged;
+    }
+
+    internal void OnQualitySettingsChanged(QualitySettings settings)
+    {
+        if (fpsCapped == settings.fpsCap)
             return;
-        autoQualityEnabled.OnChange -= SetAutoSettings;
+
+        fpsCapped = settings.fpsCap;
+        if (settings.fpsCap)
+            controller = new AutoQualityCappedFPSController(30, currentQualityIndex, qualitySettings);
```

Comment:
It would be fine to extract the value 30 as a constant with an identifying name.

Reply:
Oopsie, I forgot about that. Done